tests/e2e/App.spec.ts

```diff
@@ -0,0 +1,306 @@
+  await page.getByLabel("Command input").fill("view");
+  await page.getByRole("button", { name: "Submit Command" }).click();
+
+  await expect(page.locator('[data-index="history-command-0"]')).toContainText(
+    "Error: dataset not loaded for view"
+  );
+});
+
+test("i can load a CSV file and search existent data with column INDEX, without viewing it", async ({
+  page,
+}) => {
+  await page.getByLabel("Command input").fill("load_file path2");
+  await page.getByRole("button", { name: "Submit Command" }).click();
+
+  await expect(page.locator('[data-index="history-command-0"]')).toContainText(
+    "Loaded dataset from path2"
+  );
+
+  await page.getByLabel("Command input").fill("search 2 Data6");
+  await page.getByRole("button", { name: "Submit Command" }).click();
+
+  await expect(page.locator('[data-index="history-command-1"]')).toContainText(
+    "Data4 Data5 Data6"
+  );
+  await expect(page.locator('[data-index="history-command-2"]')).toContainText(
+    "Data4 Data7 Data6"
+  );
+});
+
+test("i can load a CSV file and search existent data with column NAME, without viewing it", async ({
+  page,
+}) => {
+  await page.getByLabel("Command input").fill("load_file path2");
+  await page.getByRole("button", { name: "Submit Command" }).click();
+
+  await expect(page.locator('[data-index="history-command-0"]')).toContainText(
+    "Loaded dataset from path2"
+  );
+
+  await page.getByLabel("Command input").fill("search Header3 Data6");
+  await page.getByRole("button", { name: "Submit Command" }).click();
+
+  await expect(page.locator('[data-index="history-command-1"]')).toContainText(
+    "Data4 Data5 Data6"
+  );
+  await expect(page.locator('[data-index="history-command-2"]')).toContainText(
+    "Data4 Data7 Data6"
+  );
+});
+
+test("i can load a CSV file and search existent data with WRONG column name, without viewing it", async ({
+  page,
+}) => {
+  await page.getByLabel("Command input").fill("load_file path2");
+  await page.getByRole("button", { name: "Submit Command" }).click();
+
+  await expect(page.locator('[data-index="history-command-0"]')).toContainText(
+    "Loaded dataset from path2"
+  );
+
+  await page.getByLabel("Command input").fill("search Header2 Data6");
+  await page.getByRole("button", { name: "Submit Command" }).click();
+
+  await expect(page.locator('[data-index="history-command-1"]')).toContainText(
+    "No results found for"
+  );
+});
+
+test("i get a useful error if i search unloaded data", async ({ page }) => {
+  await page.getByLabel("Command input").fill("search 1 hi");
+  await page.getByRole("button", { name: "Submit Command" }).click();
+
+  await expect(page.locator('[data-index="history-command-0"]')).toContainText(
+    "Error: dataset not loaded for search"
+  );
+});
+
+test("i can do EVERYTHING: view and search before loading (gets errors), load a nonexistent CSV file, load an existing file, view it, and search existent AND nonexistent data with column name and index", async ({
```

Comment:
These are all great cases, but it would be nice to seperate them into different tests since they account for different behavior.